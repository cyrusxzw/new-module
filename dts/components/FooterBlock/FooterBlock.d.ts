export function FooterBlock({
  className,
  copy,
  heading,
  headingClassName,
  isVisibleOnTabletAndMobile,
  links,
  listClassName,
  listItemClassName,
  theme,
}: {
  className: any;
  copy: any;
  heading: any;
  headingClassName: any;
  isVisibleOnTabletAndMobile: any;
  links: any;
  listClassName: any;
  listItemClassName: any;
  theme: any;
}): JSX.Element;
export namespace FooterBlock {
  namespace propTypes {
    const className: PropTypes.Requireable<string>;
    const copy: PropTypes.Requireable<PropTypes.ReactNodeLike>;
    const heading: PropTypes.Requireable<string>;
    const headingClassName: PropTypes.Requireable<string>;
    const isVisibleOnTabletAndMobile: PropTypes.Requireable<boolean>;
    const links: PropTypes.Requireable<any[]>;
    const listClassName: PropTypes.Requireable<string>;
    const listItemClassName: PropTypes.Requireable<string>;
    const theme: PropTypes.Requireable<string>;
  }
  namespace defaultProps {
    const className_1: any;
    export { className_1 as className };
    const copy_1: any;
    export { copy_1 as copy };
    const heading_1: any;
    export { heading_1 as heading };
    const headingClassName_1: any;
    export { headingClassName_1 as headingClassName };
    const isVisibleOnTabletAndMobile_1: boolean;
    export { isVisibleOnTabletAndMobile_1 as isVisibleOnTabletAndMobile };
    const links_1: any;
    export { links_1 as links };
    const listClassName_1: any;
    export { listClassName_1 as listClassName };
    const listItemClassName_1: any;
    export { listItemClassName_1 as listItemClassName };
    const theme_1: string;
    export { theme_1 as theme };
  }
}
import PropTypes from 'prop-types';
