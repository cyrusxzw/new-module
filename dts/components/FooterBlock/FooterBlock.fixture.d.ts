export namespace FooterBlockFixture {
  const actionLabel: string;
  namespace block {
    const heading: string;
    const copy: string;
  }
  namespace blockNoLink {
    const heading_1: string;
    export { heading_1 as heading };
    const copy_1: string;
    export { copy_1 as copy };
  }
  namespace navigation {
    const heading_2: string;
    export { heading_2 as heading };
    export const links: {
      text: string;
      id: string;
      url: string;
      openInANewWindow: boolean;
    }[];
  }
}
