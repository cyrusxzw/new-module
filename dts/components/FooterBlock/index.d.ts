import { FooterBlock } from './FooterBlock';
export { FooterBlock };
