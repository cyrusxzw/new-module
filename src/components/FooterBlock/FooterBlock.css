@import 'styles/index';

.base {
  font-family: var(--font-family-sans-serif-regular);

  &.dark {
    color: var(--color-dark-copy);
  }

  &.light {
    color: var(--color-light-copy);
  }
}

.heading {
  margin-top: 0;
  margin-bottom: 0;
  font-size: rem(15px);
  line-height: 2;

  @media (--viewport-sm-only) {
    font-size: rem(13px);
  }
}

.divider {
  border-bottom: 1px solid var(--color-light-copy);
  margin: 16px 0;

  &.dark {
    border-color: var(--color-dark-copy);
  }

  &.light {
    border-color: var(--color-light-copy);
  }

  @media (--viewport-sm-only) {
    margin: 6px 0;
  }
}

.list {
  padding-left: 0;
  margin: 0;
  list-style: none;
}

.copy,
.listItem {
  font-size: rem(14px);
  line-height: 2.5;
  @media (--viewport-sm-only) {
    font-size: rem(12px);
  }
  & svg {
    vertical-align: middle;
  }
}

.copy {
  margin-top: 10px;
  line-height: 1.8;
  @media (--viewport-md) {
    margin-top: 21px;
  }
  @media (--viewport-lg) {
    margin-top: 20px;
  }
  & > :last-child {
    margin-bottom: 0;
  }
}

.learnMore {
  text-decoration: underline;
}
