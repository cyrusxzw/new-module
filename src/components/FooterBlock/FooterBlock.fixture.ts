const FooterBlockFixture = {
  actionLabel: 'Learn More',
  block: {
    heading: 'Sustainability',
    copy:
      'Our approach to sustainability is guided by where we can have the greatest impact. We have ceritifications from Crulety Free International as well as B Corp Certifications',
  },
  blockNoLink: {
    heading: 'Acknowledgment of Country ',
    copy:
      'Aesop acknowledges the Traditional Owners of the lands on which we live and work, and pays deepest respects to Elders past, present and emerging. We also acknowledge that sovereignty of these lands has never been ceded.',
  },
  navigation: {
    heading: 'Orders & Support',
    links: [
      {
        text: 'Contact us',
        id: 'contact-us',
        url: 'https://www.aesop.com/au/r/contact-us',
        openInANewWindow: true,
      },
      {
        text: 'Delivery and returns',
        id: 'delivery-and-returns',
        url: 'https://www.aesop.com/au/r/shipping/',
        openInANewWindow: true,
      },
      {
        text: 'Click And Collect',
        id: 'click-and-collect',
        url: 'https://www.aesop.com/au/r/contact-us',
        openInANewWindow: true,
      },
      {
        text: 'Track Your Orders',
        id: 'track-your-orders',
        url: 'https://www.aesop.com/au/r/shipping/',
        openInANewWindow: false,
      },
    ],
  },
};

export { FooterBlockFixture };
