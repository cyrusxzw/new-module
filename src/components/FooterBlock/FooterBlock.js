import React from 'react';
import PropTypes from 'prop-types';
import { Heading } from '~/components/Heading';
import { Hidden } from '~/components/Hidden';
import { Hyperlink } from '~/components/Hyperlink';
import { HYPERLINK_STYLE_TYPES } from '~/constants';
import cx from 'classnames';
import styles from './FooterBlock.css';

const {
  EXTERNAL_TEXT_LINK,
  INTERNAL_NO_ICON_TEXT_LINK,
} = HYPERLINK_STYLE_TYPES;

const FooterBlock = ({
  className,
  copy,
  heading,
  headingClassName,
  isVisibleOnTabletAndMobile,
  links,
  listClassName,
  listItemClassName,
  theme,
}) => {
  const baseClassSet = cx(styles.base, styles[theme], className);
  const copyClassSet = cx(styles.listItem, styles.copy);
  const dividerClassSet = cx(styles.divider, styles[theme]);
  const headingClassSet = cx(styles.heading, headingClassName);
  const listClassSet = cx(styles.list, listClassName);
  const listItemClassSet = cx(styles.listItem, listItemClassName);

  return (
    <Hidden
      isMedium={!isVisibleOnTabletAndMobile}
      isSmall={!isVisibleOnTabletAndMobile}
    >
      <section className={baseClassSet}>
        {heading && (
          <Heading
            className={headingClassSet}
            hasMediumWeightFont={true}
            level="3"
            size="xXSmall"
            theme="light"
          >
            {heading}
          </Heading>
        )}
        <div className={dividerClassSet} />
        {links?.length && (
          <ul className={listClassSet}>
            {links.map(link => (
              <li key={link.id}>
                <Hyperlink
                  className={listItemClassSet}
                  dataTestRef="FOOTER_LINK"
                  hasTargetInNewWindow={link.openInANewWindow}
                  style={
                    link.openInANewWindow
                      ? EXTERNAL_TEXT_LINK
                      : INTERNAL_NO_ICON_TEXT_LINK
                  }
                  theme={theme}
                  title={link.text}
                  url={link.url}
                >
                  {link.text}
                </Hyperlink>
              </li>
            ))}
          </ul>
        )}
        {copy && <div className={copyClassSet}>{copy}</div>}
      </section>
    </Hidden>
  );
};

FooterBlock.propTypes = {
  className: PropTypes.string,
  copy: PropTypes.node,
  heading: PropTypes.string,
  headingClassName: PropTypes.string,
  isVisibleOnTabletAndMobile: PropTypes.bool,
  links: PropTypes.array,
  listClassName: PropTypes.string,
  listItemClassName: PropTypes.string,
  theme: PropTypes.oneOf(['dark', 'light']),
};

FooterBlock.defaultProps = {
  className: undefined,
  copy: undefined,
  heading: undefined,
  headingClassName: undefined,
  isVisibleOnTabletAndMobile: true,
  links: undefined,
  listClassName: undefined,
  listItemClassName: undefined,
  theme: 'dark',
};

export { FooterBlock };
