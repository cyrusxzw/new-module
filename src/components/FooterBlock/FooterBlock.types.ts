import type { ReactNode } from 'react';
import type { ComponentWithoutChildren, Themes } from '~/types';

type Link = {
  id?: string;
  openInANewWindow?: boolean;
  text: string;
  url: string;
};

type FooterBlockProps = {
  className?: string;
  copy?: ReactNode;
  heading?: string;
  headingClassName?: string;
  isVisibleOnTabletAndMobile?: boolean;
  links?: Link[];
  listClassName?: string;
  listItemClassName?: string;
  theme?: Themes;
};

export type FooterBlockType = ComponentWithoutChildren<FooterBlockProps>;
