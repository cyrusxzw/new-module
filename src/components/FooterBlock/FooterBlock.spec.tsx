import React from 'react';
import { render } from '@testing-library/react';
import { FooterBlock } from './FooterBlock';
import { FooterBlockFixture } from './FooterBlock.fixture';

describe('<FooterBlock />', () => {
  it('should render base component correctly', () => {
    const { container } = render(
      <FooterBlock copy={FooterBlockFixture.blockNoLink.copy} theme="dark" />,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders block navigation component correctly', () => {
    const { container } = render(
      <FooterBlock
        isVisibleOnTabletAndMobile={false}
        links={FooterBlockFixture.navigation.links}
        theme="dark"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
