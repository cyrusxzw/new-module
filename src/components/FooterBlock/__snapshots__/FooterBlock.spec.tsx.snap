// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FooterBlock /> renders block navigation component correctly 1`] = `
<div>
  <section
    class="base dark"
  >
    <div
      class="divider dark"
    />
    <ul
      class="list"
    >
      <li>
        <a
          class="base hasIcon hasExternalIcon left dark listItem"
          data-test-ref="FOOTER_LINK"
          href="https://www.aesop.com/au/r/contact-us"
          target="_blank"
          title="Contact us"
        >
          Contact us
           
          <i
            aria-hidden="true"
            class="icon"
          >
            <svg
              aria-labelledby="rightUpArrow-1"
              class="base rightUpArrow dark"
              data-testid="data-testid-Icon"
              focusable="false"
              height="12"
              role="img"
              style="height: 12px; width: 12px;"
              tabindex="-1"
              viewBox="0 0 50 50"
              width="12"
            >
              <title
                id="rightUpArrow-1"
              >
                rightUpArrow
              </title>
              <g>
                <path
                  d="M21.8,0.2l28.1-0.1v28.1l-4.5,0l0.1-20.5L16.2,37.1l-3.4-3.4L42.1,4.4L21.8,4.4L21.8,0.2L21.8,0.2z"
                />
              </g>
            </svg>
          </i>
        </a>
      </li>
      <li>
        <a
          class="base hasIcon hasExternalIcon left dark listItem"
          data-test-ref="FOOTER_LINK"
          href="https://www.aesop.com/au/r/shipping/"
          target="_blank"
          title="Delivery and returns"
        >
          Delivery and returns
           
          <i
            aria-hidden="true"
            class="icon"
          >
            <svg
              aria-labelledby="rightUpArrow-2"
              class="base rightUpArrow dark"
              data-testid="data-testid-Icon"
              focusable="false"
              height="12"
              role="img"
              style="height: 12px; width: 12px;"
              tabindex="-1"
              viewBox="0 0 50 50"
              width="12"
            >
              <title
                id="rightUpArrow-2"
              >
                rightUpArrow
              </title>
              <g>
                <path
                  d="M21.8,0.2l28.1-0.1v28.1l-4.5,0l0.1-20.5L16.2,37.1l-3.4-3.4L42.1,4.4L21.8,4.4L21.8,0.2L21.8,0.2z"
                />
              </g>
            </svg>
          </i>
        </a>
      </li>
      <li>
        <a
          class="base hasIcon hasExternalIcon left dark listItem"
          data-test-ref="FOOTER_LINK"
          href="https://www.aesop.com/au/r/contact-us"
          target="_blank"
          title="Click And Collect"
        >
          Click And Collect
           
          <i
            aria-hidden="true"
            class="icon"
          >
            <svg
              aria-labelledby="rightUpArrow-3"
              class="base rightUpArrow dark"
              data-testid="data-testid-Icon"
              focusable="false"
              height="12"
              role="img"
              style="height: 12px; width: 12px;"
              tabindex="-1"
              viewBox="0 0 50 50"
              width="12"
            >
              <title
                id="rightUpArrow-3"
              >
                rightUpArrow
              </title>
              <g>
                <path
                  d="M21.8,0.2l28.1-0.1v28.1l-4.5,0l0.1-20.5L16.2,37.1l-3.4-3.4L42.1,4.4L21.8,4.4L21.8,0.2L21.8,0.2z"
                />
              </g>
            </svg>
          </i>
        </a>
      </li>
      <li>
        <a
          class="base left dark listItem"
          data-test-ref="FOOTER_LINK"
          href="https://www.aesop.com/au/r/shipping/"
          target="_self"
          title="Track Your Orders"
        >
          Track Your Orders
        </a>
      </li>
    </ul>
  </section>
</div>
`;

exports[`<FooterBlock /> should render base component correctly 1`] = `
<div>
  <section
    class="base dark"
  >
    <div
      class="divider dark"
    />
    <div
      class="listItem copy"
    >
      Aesop acknowledges the Traditional Owners of the lands on which we live and work, and pays deepest respects to Elders past, present and emerging. We also acknowledge that sovereignty of these lands has never been ceded.
    </div>
  </section>
</div>
`;
